src/main.ts

```diff
@@ -50,15 +50,23 @@ function makeWriter(): CodeWriter {
   };
 }
 
+type NameFunction = (originalName: string) => string;
+
 export interface GenerateOptions {
   format?: boolean;
   formatOptions?: PrettierOptions;
   includeImport?: boolean;
+  includeTypes?: boolean;
+  formatRuntypeName?: NameFunction;
+  formatTypeName?: NameFunction;
 }
 
 const defaultOptions: GenerateOptions = {
   format: true,
   includeImport: true,
+  includeTypes: true,
+  formatRuntypeName: (e) => e,
```

Comment:
I like the lowercase vs uppercase default behavior, but what if the name is already uppercased? Might make sense to do something like this: 
```suggestion
  formatRuntypeName: (e) => e[0].toLowerCase() + e.slice(1),
```

Reply:
Right. Sounds reasonable. People will likely provide their own formatters for any fancy namings anyway. Like, this wont work if the name starts with `_` for example, but I don't think that's important enough to care about